packages/react-native-executorch/common/rnexecutorch/models/ocr/RecognitionHandlerUtils.cpp

```diff
@@ -0,0 +1,150 @@
+#include "RecognitionHandlerUtils.h"
+#include <algorithm>
+#include <rnexecutorch/data_processing/ImageProcessing.h>
+#include <rnexecutorch/models/ocr/Constants.h>
+
+namespace rnexecutorch::ocr {
+PaddingInfo calculateResizeRatioAndPaddings(cv::Size size,
+                                            cv::Size desiredSize) {
+  float newRatioH = static_cast<float>(desiredSize.height) / size.height;
+  float newRatioW = static_cast<float>(desiredSize.width) / size.width;
+  float resizeRatio = std::min(newRatioH, newRatioW);
+
+  int32_t newHeight = static_cast<int32_t>(size.height * resizeRatio);
+  int32_t newWidth = static_cast<int32_t>(size.width * resizeRatio);
+
+  int32_t deltaH = desiredSize.height - newHeight;
+  int32_t deltaW = desiredSize.width - newWidth;
+
+  int32_t top = deltaH / 2;
+  int32_t left = deltaW / 2;
+
+  float heightRatio = static_cast<float>(size.height) / desiredSize.height;
+  float widthRatio = static_cast<float>(size.width) / desiredSize.width;
+
+  resizeRatio = std::max(heightRatio, widthRatio);
+  return PaddingInfo{resizeRatio, top, left};
+}
+
+void computeRatioAndResize(cv::Mat &img, cv::Size size, int32_t modelHeight) {
+  double ratio =
+      static_cast<double>(size.width) / static_cast<double>(size.height);
+  cv::Size resizedSize;
+  if (ratio < 1.0) {
+    ratio = 1.0 / ratio;
+    resizedSize =
+        cv::Size(modelHeight, static_cast<int32_t>(modelHeight * ratio));
+  } else {
+    resizedSize =
+        cv::Size(static_cast<int32_t>(modelHeight * ratio), modelHeight);
+  }
+  cv::resize(img, img, resizedSize, 0.0, 0.0, cv::INTER_LANCZOS4);
+}
+
+cv::Mat cropImage(DetectorBBox box, cv::Mat &image, int32_t modelHeight) {
+  // Convert custom points to cv::Point2f
+  std::array<cv::Point2f, 4> points;
+  for (size_t i = 0; i < 4; ++i) {
+    points[i] = cv::Point2f(box.bbox[i].x, box.bbox[i].y);
+  }
+
+  cv::RotatedRect rotatedRect = cv::minAreaRect(points);
+  cv::Point2f rectPoints[4];
+  rotatedRect.points(rectPoints);
+
+  // Rotate the image
+  cv::Point2f imageCenter(image.cols / 2.0f, image.rows / 2.0f);
+  cv::Mat rotationMatrix = cv::getRotationMatrix2D(imageCenter, box.angle, 1.0);
+  cv::Mat rotatedImage;
+  cv::warpAffine(image, rotatedImage, rotationMatrix, image.size(),
+                 cv::INTER_LINEAR);
+
+  cv::Mat rectMat(4, 2, CV_32FC2);
+  for (int i = 0; i < 4; ++i) {
+    rectMat.at<cv::Vec2f>(i, 0) = cv::Vec2f(rectPoints[i].x, rectPoints[i].y);
+  }
```

Comment:
I'm not sure if this constructor will work here, so you need to check, but unrolling is obligatory ;)
```suggestion
#pragma unroll
  for (int i = 0; i < 4; ++i) {
    rectMat.at<cv::Vec2f>(i, 0) = {rectPoints[i].x, rectPoints[i].y};
  }
```